packages/react-router/src/components/FocusContextProvider/FocusContextProvider.tsx

```diff
@@ -0,0 +1,27 @@
+import React, {useRef, useEffect, ReactNode} from 'react';
+import {FocusContext} from '../../context';
+
+export function FocusContextProvider({children}: {children: ReactNode}) {
+  const focusRef = React.createRef<HTMLElement>();
+  const focus = () => {
+    return focusRef?.current?.focus();
+  };
+
+  const firstRender = useRef(true);
+
+  useEffect(() => {
+    if (!firstRender.current) {
+      focus();
+    }
+  });
+
+  useEffect(() => {
+    firstRender.current = false;
+  }, []);
+
+  return (
+    <FocusContext.Provider value={{focusRef, focus}}>
```

Comment:
In general, it's best to avoid constructing a new object for context every time, as it requires all consumers of that context in the entire subtree to be re-rendered. In this case, though, I don't think you need to pass `focus` around at all?

Reply:
Yes, good catch, no need to pass the focus around!